compiler/make/Makefile.compiler.head

```diff
@@ -53,6 +53,10 @@ INCL_CFLAGS = -I. \
 
 COMP_CXXFLAGS += $(INCL_CFLAGS)
 
+ifdef CHPL_DEVELOPER
+COMP_CXXFLAGS += -Wmissing-braces
```

Comment:
Is this the right place for this flag or should be in Makefile.gnu/Makefile.clang with `WARN_CXXFLAGS` or something?

Reply:
Ahh thanks I knew there was probably a better place. I guess add to Makefile.{gnu,clang} for now as those are probably the most likely to be used by developers?

Reply:
Yeah, and at least in practice it's very rare for anybody to build the compiler with anything but gnu/clang. Even on cray's when building the runtime with cce/intel we still build the compiler with gnu.

Reply:
moved into those makefiles